zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/listeners/task/TaskListenerTest.java

```diff
@@ -0,0 +1,496 @@
+        .satisfies(
+            recordValue -> {
+              assertThat(recordValue.getAction()).isEqualTo("my_custom_action");
+              assertThat(recordValue.getVariables()).containsExactly(entry("foo_var", "bar"));
+            });
+
+    assertThatProcessInstanceCompleted(processInstanceKey);
+  }
+
+  @Test
+  public void shouldRetryTaskListenerWhenListenerJobFailed() {
+    // given
+    final long processInstanceKey =
+        createProcessInstance(
+            createProcessWithCompleteTaskListeners(LISTENER_TYPE, LISTENER_TYPE + "_2"));
+
+    ENGINE.userTask().ofInstance(processInstanceKey).complete();
+
+    // when: fail listener job with retries
+    ENGINE.job().ofInstance(processInstanceKey).withType(LISTENER_TYPE).withRetries(1).fail();
+    // complete failed and remaining listeners job
+    completeJobs(processInstanceKey, LISTENER_TYPE, LISTENER_TYPE + "_2");
+
+    // then: assert the listener job was completed after the failure
+    assertThat(records().betweenProcessInstance(processInstanceKey))
+        .extracting(Record::getValueType, Record::getIntent)
+        .containsSubsequence(
+            tuple(ValueType.USER_TASK, UserTaskIntent.COMPLETING),
+            tuple(ValueType.JOB, JobIntent.CREATED),
+            tuple(ValueType.JOB, JobIntent.FAILED),
+            tuple(ValueType.JOB, JobIntent.COMPLETE),
+            tuple(ValueType.JOB, JobIntent.COMPLETED),
+            tuple(ValueType.USER_TASK, UserTaskIntent.COMPLETE_TASK_LISTENER),
+            tuple(ValueType.JOB, JobIntent.CREATED),
+            tuple(ValueType.JOB, JobIntent.COMPLETE),
+            tuple(ValueType.JOB, JobIntent.COMPLETED),
+            tuple(ValueType.USER_TASK, UserTaskIntent.COMPLETE_TASK_LISTENER),
+            tuple(ValueType.USER_TASK, UserTaskIntent.COMPLETED));
+
+    assertThatProcessInstanceCompleted(processInstanceKey);
+  }
+
+  @Test
+  public void shouldCreateIncidentForListenerWhenNoRetriesLeftAndProceedWithRemainingListeners() {
+    // given
+    final long processInstanceKey =
+        createProcessInstance(
+            createProcessWithCompleteTaskListeners(
+                LISTENER_TYPE, LISTENER_TYPE + "_2", LISTENER_TYPE + "_3"));
+
+    ENGINE.userTask().ofInstance(processInstanceKey).complete();
+    completeJobs(processInstanceKey, LISTENER_TYPE);
+
+    // when: fail 2nd listener job with no retries
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(LISTENER_TYPE + "_2")
+        .withRetries(0)
+        .fail();
+
+    // then: incident created
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+    Assertions.assertThat(incident.getValue())
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasErrorType(ErrorType.JOB_NO_RETRIES)
```

Comment:
~~❌ I believe we decided to use `TASK_LISTENER_NO_RETRIES`~~

EDIT: It seems we'll tackle this elsewhere: 
- https://github.com/camunda/camunda/issues/23492